extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java

```diff
@@ -37,28 +41,50 @@
     /** RegEx to extract a scale definition */
     private static final Pattern LIMITS_PATTERN = Pattern.compile("(\\[|\\])(.*)\\.\\.(.*)(\\[|\\])");
 
+    class OrderedProperties extends Properties {
+        private static final long serialVersionUID = 3860553217028220119L;
+        private final HashSet<Object> keys = new LinkedHashSet<Object>();
+
+        public OrderedProperties() {
+        }
+
+        public Iterable<Object> orderedKeys() {
+            return Collections.list(keys());
+        }
+
+        @Override
+        public Enumeration<Object> keys() {
+            return Collections.<Object> enumeration(keys);
```

Comment:
Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`

Reply:
Diamond operator does not work here. So left it as is.

Reply:
Hmm, sorry then Intellij had some other complaint about that line but its not that important :-)